File: GTTS/GTTS.Web/ClientApp/src/app/models/INL.UnitLibraryService.Models/unit_item.ts


import { GetUnitAlias_Item } from './get-unit-alias_item';
import { GetUnitLocation_Item } from './get-unit-location_item';
import { GetUnitCommander_Item } from './get-unit-commander_item';
import { IUnit_Item } from './iunit_item';

export class Unit_Item implements IUnit_Item {
  
	public UnitID: number = 0;
	public UnitParentID?: number;
	public UnitParentName: string = "";
	public UnitParentNameEnglish: string = "";
	public AgencyName: string = "";
	public AgencyNameEnglish: string = "";
	public UnitAgencyName: string = "";
	public UnitParents: string = "";
	public UnitBreakdown: string = "";
	public UnitBreakdownLocalLang: string = "";
	public CountryID: number = 0;
	public CountryName: string = "";
	public UnitLocationID?: number;
	public ConsularDistrictID?: number;
	public UnitName: string = "";
	public UnitNameEnglish: string = "";
	public IsMainAgency: boolean = false;
	public UnitMainAgencyID?: number;
	public UnitAcronym: string = "";
	public UnitGenID: string = "";
	public UnitTypeID: number = 0;
	public UnitType: string = "";
	public GovtLevelID?: number;
	public GovtLevel: string = "";
	public UnitLevelID?: number;
	public UnitLevel: string = "";
	public VettingBatchTypeID: number = 0;
	public VettingBatchTypeCode: string = "";
	public VettingActivityTypeID: number = 0;
	public VettingActivityType: string = "";
	public ReportingTypeID?: number;
	public ReportingType: string = "";
	public UnitHeadPersonID?: number;
	public CommanderFirstName: string = "";
	public CommanderLastName: string = "";
	public UnitHeadJobTitle: string = "";
	public UnitHeadRankID?: number;
	public RankName: string = "";
	public UnitHeadRank: string = "";
	public UnitHeadFirstMiddleNames: string = "";
	public UnitHeadLastNames: string = "";
	public UnitHeadIDNumber: string = "";
	public UnitHeadGender?: string;
	public UnitHeadDOB?: Date;
	public UnitHeadPoliceMilSecID: string = "";
	public UnitHeadPOBCityID?: number;
	public UnitHeadResidenceCityID?: number;
	public UnitHeadContactEmail: string = "";
	public UnitHeadContactPhone: string = "";
	public UnitHeadHighestEducationID?: number;
	public UnitHeadEnglishLanguageProficiencyID?: number;
	public HQLocationID?: number;
	public POCName: string = "";
	public POCEmailAddress: string = "";
	public POCTelephone: string = "";
	public VettingPrefix: string = "";
	public HasDutyToInform: boolean = false;
	public IsLocked: boolean = false;
	public IsActive?: boolean;
	public ModifiedByAppUserID: number = 0;
	public ModifiedDate: Date = new Date(0);
	public UnitParentJson: string = "";
	public CountryJson: string = "";
	public LocationJson: string = "";
	public PostJson: string = "";
	public MainAgencyJson: string = "";
	public HQLocationJson: string = "";
	public UnitAlias?: GetUnitAlias_Item[];
	public UnitLocation?: GetUnitLocation_Item;
	public HQLocation?: GetUnitLocation_Item;
	public Commander?: GetUnitCommander_Item;
  
}

File: GTTS/INL.UnitLibraryService.Models/Unit_Item.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace INL.UnitLibraryService.Models
{
    public class Unit_Item : IUnit_Item
    {
        public long UnitID { get; set; }
        public long? UnitParentID { get; set; }
        public string UnitParentName { get; set; }
        public string UnitParentNameEnglish { get; set; }
        public string AgencyName { get; set; }
        public string AgencyNameEnglish { get; set; }
        public string UnitAgencyName { get; set; }
        public string UnitParents { get; set; }
        public string UnitBreakdown { get; set; }
        public string UnitBreakdownLocalLang { get; set; }
        public int CountryID { get; set; }
        public string CountryName { get; set; }
        public long? UnitLocationID { get; set; }
        public int? ConsularDistrictID { get; set; }
        public string UnitName { get; set; }
        public string UnitNameEnglish { get; set; }
        public bool IsMainAgency { get; set; }
        public long? UnitMainAgencyID { get; set; }
        public string UnitAcronym { get; set; }
        public string UnitGenID { get; set; }
        public int UnitTypeID { get; set; }
        public string UnitType { get; set; }
        public int? GovtLevelID { get; set; }
        public string GovtLevel { get; set; }
        public int? UnitLevelID { get; set; }
        public string UnitLevel { get; set; }
        public byte VettingBatchTypeID { get; set; }
        public string VettingBatchTypeCode { get; set; }
        public int VettingActivityTypeID { get; set; }
        public string VettingActivityType { get; set; }
        public int? ReportingTypeID { get; set; }
        public string ReportingType { get; set; }
        public long? UnitHeadPersonID { get; set; }
        public string CommanderFirstName { get; set; }
        public string CommanderLastName { get; set; }
        public string UnitHeadJobTitle { get; set; }
        public int? UnitHeadRankID { get; set; }
        public string RankName { get; set; }
        public string UnitHeadRank { get; set; }
        public string UnitHeadFirstMiddleNames { get; set; }
        public string UnitHeadLastNames { get; set; }
        public string UnitHeadIDNumber { get; set; }
        public char? UnitHeadGender { get; set; }
        public DateTime? UnitHeadDOB { get; set; }
        public string UnitHeadPoliceMilSecID { get; set; }
        public int? UnitHeadPOBCityID { get; set; }
        public int? UnitHeadResidenceCityID { get; set; }
        public string UnitHeadContactEmail { get; set; }
        public string UnitHeadContactPhone { get; set; }
        public int? UnitHeadHighestEducationID { get; set; }
        public int? UnitHeadEnglishLanguageProficiencyID { get; set; }
        public long? HQLocationID { get; set; }
        public string POCName { get; set; }
        public string POCEmailAddress { get; set; }
        public string POCTelephone { get; set; }
        public string VettingPrefix { get; set; }
        public bool HasDutyToInform { get; set; }
        public bool IsLocked { get; set; }
        public bool? IsActive { get; set; }
        public int ModifiedByAppUserID { get; set; }
        public DateTime ModifiedDate { get; set; }
        public string UnitParentJson { get; set; }
        public string CountryJson { get; set; }
        public string LocationJson { get; set; }
        public string PostJson { get; set; }
        public string MainAgencyJson { get; set; }
        public string HQLocationJson { get; set; }

        public List<GetUnitAlias_Item> UnitAlias { get; set; }
        public GetUnitLocation_Item UnitLocation { get; set; }
        public GetUnitLocation_Item HQLocation { get; set; }
        public GetUnitCommander_Item Commander { get; set; }
    }
}
